deps/spidershim/test/value.cc

```diff
@@ -778,6 +778,24 @@ TEST(SpiderShim, Date) {
   EXPECT_EQ(time, Date::Cast(*date.ToLocalChecked()->ToObject())->ValueOf());
 }
 
+namespace {
+
+class TestExternalStringResource : public String::ExternalStringResource {
+  public:
+    TestExternalStringResource(const uint16_t* source, size_t length)
+        : data_(source), length_(length) {}
+    ~TestExternalStringResource() {
+      // XXX Check that we get called when the string is finalized.
```

Comment:
> you can pass in a pointer to a global bool or something like that, and add an atexit callback that checks whether that bool has been set as a result of this finalizer being called. That would be some dark magic but I think it should work reliably.

That works in theory, except that I don't see a way to hook an atexit check into the gtest harness. Of course I could simply *printf* the result, but if I'm only going to do that, then I might as well simply *printf* inside the destructor itself (which is what I do when manually testing this branch).

Perhaps instead I can define a TEST(SpiderShim, ExternalStringDestructor) function that checks the result of the destructor. The test harness appears to run the tests in deterministic order (and reconstruct the engine, or at least the isolate, for each test function), so that should work.